bc_obps/reporting/service/report_new_entrant_service.py

```diff
@@ -0,0 +1,182 @@
+        result = [
+            {
+                "name": "basic",
+                "title": "Emission categories after new entrant period began",
+                "emissionData": emissions_by_type["basic"],
+            },
+            {
+                "name": "fuel_excluded",
+                "title": "Emissions excluded by fuel type",
+                "emissionData": emissions_by_type["fuel_excluded"],
+            },
+        ]
+        if naics_code == "324110" and emissions_by_type["other_excluded"]:
+            result.append(
+                {
+                    "name": "other_excluded",
+                    "title": "Other excluded emissions",
+                    "emissionData": emissions_by_type["other_excluded"],
+                }
+            )
+
+        return result
+
+    @classmethod
+    @transaction.atomic
+    def save_new_entrant_data(
+        cls,
+        report_version_id: int,
+        data: ReportNewEntrantSchemaIn,
+    ) -> None:
+        """Saves new entrant data and updates the corresponding records."""
+        report_version = ReportVersion.objects.get(pk=report_version_id)
+
+        report_new_entrant, _ = ReportNewEntrant.objects.update_or_create(
+            report_version=report_version,
+            defaults={
+                "authorization_date": data.authorization_date,
+                "first_shipment_date": data.first_shipment_date,
+                "new_entrant_period_start": data.new_entrant_period_start,
+                "assertion_statement": data.assertion_statement,
+            },
+        )
+
+        valid_emission_data = [
+            emission_item
+            for category in data.emissions
+            for emission_item in category["emissionData"]
+            if emission_item.get("emission") is not None
+        ]
+
+        for emission_item in valid_emission_data:
+            ReportNewEntrantEmissions.objects.update_or_create(
+                report_new_entrant=report_new_entrant,
+                emission_category_id=emission_item["id"],
+                defaults={"emission": emission_item["emission"]},
+            )
+
+        emission_ids_with_data = [item["id"] for item in valid_emission_data]
+        ReportNewEntrantEmissions.objects.filter(report_new_entrant=report_new_entrant).exclude(
+            emission_category_id__in=emission_ids_with_data
+        ).delete()
+
+        valid_productions = [
+            ReportNewEntrantProduction(
+                report_new_entrant=report_new_entrant,
+                product_id=product["id"],
+                production_amount=product["production_amount"],
+            )
+            for product in data.products
+            if product.get("production_amount") is not None
+        ]
+
+        valid_product_ids = [prod.product_id for prod in valid_productions]
+
+        ReportNewEntrantProduction.objects.filter(report_new_entrant=report_new_entrant).exclude(
+            product_id__in=valid_product_ids
+        ).delete()
```

Comment:
We can probably delete all the records before saving the new ones, since the form submits everything at once, then we don't need to do some juggling with existing IDs